lib/cli_executor.rb

```diff
@@ -65,7 +65,7 @@ def self.get_version_for(user, host)
       fake_config.unlink
 
       raise "cannot execute on host #{host.hostname} as user '#{user}'" unless res[:success]
-      return res[:response].scan(/^os?c v(.+)$|GitVersion:"v([^"]+)"/)[0].find{|v| v != nil}
+      return res[:response].scan(/^os?c v(.+)$|GitVersion:"v([^"]+)"|^Client Version: v(.+)$/)[0].find{|v| v != nil}
```

Comment:
Can we only look for `GitVersion` and `Client Version:`? I see in 4.1 we need `GitVersion`, while in 4.2 `Client Version`.

Reply:
Do you mean not to look for "^os?c v(.+)$"? It's for 3.x, I thinks we still need it for 3.x auto.
Or you mean we look for GitVersion and Client Version instead of GitVersion:"v([^"]+)" and ^Client Version: v(.+)$ ?